pkg/controllers/management/auth/auth_config_test.go

```diff
@@ -104,10 +100,21 @@ func TestCleanupRuns(t *testing.T) {
 				Enabled: test.configEnabled,
 			}
 
+			var service cleanupService
+			controller := authConfigController{
+				cleanup:     &service,
+				authConfigs: newMockAuthConfigClient(config),
+			}
+
 			obj, err := controller.sync("test", config)
 			require.NoError(t, err)
 			assert.Equal(t, test.expectCleanup, service.cleanupCalled)
-			assert.Equal(t, test.newAnnotationValue, obj.(*v3.AuthConfig).Annotations[CleanupAnnotation])
+
+			obj, err = controller.authConfigs.Get(config.Name, metav1.GetOptions{})
+			require.NoError(t, err)
+			u, _ := obj.(runtime.Unstructured)
+			authConfig := u.UnstructuredContent()
+			assert.Equal(t, test.newAnnotationValue, authConfig["metadata"].(map[string]any)["annotations"].(map[string]any)[CleanupAnnotation])
```

Comment:
I don't think this test is making a meaningful assertion now. Before, you were testing the output of sync to make sure that the annotation was added. Now you are just asserting that your mock returns the authConfig that you constructed as part of the test. I would recommend changing this to either:

- Record the new value passed into the update call, and make assertions on that value (Don't like this option)
- Make assertions on the return of sync like before (much better option IMO)